### DISCUSSION TITLE
auth.users - customization

### DISCUSSION THREAD

**Discussion by rcsilva81** [supabase/supabase#924] [Questions] [answered]
Guys, I want to use the Auth API already in place but my requirements ask for a way to invalidate the user and also, some extra field like name and profile settings. So, is there a way to invalidate a user in place?  Also, would it be possible to add extra fields to auth.users and update those extra columns or the best approach is to create a public.users?

Thanks, 
Rob

**Comment by silentworks** [comment, marked as answer]
I think you would have to come up with your own solution to invalidate a user like by using a flag or something in your table. Having the public.users table is the best approach for additional data.

**Reply by rcsilva81** [reply]
Tks, silentworks, that's what I did.

**Comment by ChronSyn** [comment]
In terms of having a public.users table, what would be the best approach to linking a specific row to a a user in auth.users? At the moment, I'm considering having an `auth_user_id` UUID in public.users, which is identical to the UUID of a user in auth.users.

When a user signs up via a third party provider (e.g. Facebook, Github, Google) and is redirected back to my application (i.e. my app frontend), what would be the best approach to link both? It definitely doesn't feel safe to be inserting the `public.users` record as part of a `auth.signIn().then()` chain. Is there any way to do this from within the Supabase dashboard - for example, as an 'after auth' workflow?

**Reply by silentworks** [reply]
For something like this you can use a stored procedure. Take a look at the example provided by supabase themselves https://supabase.io/docs/guides/auth#create-a-publicusers-table

**Reply by ChronSyn** [reply]
> For something like this you can use a stored procedure. Take a look at the example provided by supabase themselves https://supabase.io/docs/guides/auth#create-a-publicusers-table

Fantastic! This does exactly what I want and made me realize I need to brush up on my SQL again
